http/src/main/java/io/micronaut/http/bind/binders/QueryValueArgumentBinder.java

```diff
@@ -130,6 +146,51 @@ public BindingResult<T> bind(ArgumentConversionContext<T> context, HttpRequest<?
         return doBind(context, parameters, BindingResult.unsatisfied());
     }
 
+    private BindingResult<T> bindPojo(ArgumentConversionContext<T> context,
+                                      HttpRequest<?> source,
+                                      AnnotationMetadata annotationMetadata,
+                                      ConvertibleMultiValues<String> parameters,
+                                      Argument<T> argument) {
+        Optional<BeanIntrospection<T>> introspectionOpt = BeanIntrospector.SHARED.findIntrospection(argument.getType());
+        if (introspectionOpt.isEmpty()) {
+            return BindingResult.unsatisfied();
+        }
+
+        BeanIntrospection<T> introspection = introspectionOpt.get();
+        BeanIntrospection.Builder<T> introspectionBuilder = introspection.builder();
+
+        for (BeanProperty<T, Object> property : introspection.getBeanProperties()) {
+            String propertyName = property.getName();
+            Optional<String> value = parameters.getFirst(propertyName);
+
+            if (value.isEmpty()) {
+                value = property.asArgument()
```

Comment:
remove the call to `asArgument` and just use the argument from builder arguments above